src/sip128.rs

```diff
@@ -378,13 +357,8 @@ impl SipHasher128 {
         }
     }
 
-    #[inline(always)]
-    pub fn finish128(mut self) -> [u64; 2] {
-        SipHasher128::finish128_inner(self.nbuf, &mut self.buf, self.state, self.processed)
-    }
-
     #[inline]
-    fn finish128_inner(
+    unsafe fn finish128_inner(
```

Comment:
Why is this unsafe now?

Reply:
I missed that in the other PR, but if `nbuf` is too big, it can cause UB as we will OBB `buf`.

Reply:
I wonder if turning the `debug_assert!(nbuf < BUFFER_SIZE)` into a regular `assert!(nbuf < BUFFER_SIZE)` would hurt performance.

Reply:
In any case, let's document what condition exactly must be met for a call to be safe.

Reply:
I've documented the safety condition, by copying the safety condition from another unsafe function.